### DISCUSSION TITLE
doesn't Microsoft have plans to develop its own 2D graphics library Or create a wrapper library ImpellerSharp ?

### DISCUSSION THREAD

**Discussion by sgf** [mono/SkiaSharp#2981] [General]
Google already has Skia, and for performance and optimization issues, they are still working hard to develop [Impeller](https://github.com/flutter/engine/tree/main/impeller).
So why doesn't Microsoft have plans to develop its own 2D graphics library?

Or does Microsoft have plans to create a wrapper library like "ImpellerSharp" for  Impeller?

**Comment by wieslawsoltes** [comment]


**Reply by sgf** [reply]
> Also please not this is mono repository and not Microsoft

But hasn't mono been acquired by Microsoft?

**Comment by sgf** [comment]
Does win2d or d2d support cross-platform?

**Reply by sgf** [reply]
> You did not ask about cross-platform in you question

But the 2D graphics engines from Google that I mentioned are all cross-platform.

**Comment by sgf** [comment]
I still closed this discussion, although it is friendly to .net cross-platform, mainly to support MAUI.
Supporting the MAUI.Graphics project would be a good choice. But I know that this kind of thing is not something that I, the mono team, or even the .net team can decide.
Because it depends on whether Microsoft is willing to invest in it (for example: provide financial support). As we all know, although .net is open source, it is dominated by Microsoft, including the main team and the main financial support.
Therefore, it is not very meaningful to discuss whether an independent 2D graphics library should be developed. Of course it should, but the reality does not allow it.
